CMakeLists.txt

```diff
@@ -84,89 +86,206 @@ if (MSVC)
 endif()
 
 
-if (ERT_USE_OPENMP)
-  find_package(OpenMP)
-  if (OPENMP_FOUND) 
-     message(STATUS "Enabling OpenMP support")
-     # The actual use of OpenMP is only in the libecl library - the compile flags is only applied there.
-  else()
-     set( ERT_USE_OPENMP OFF )
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Modules)
+find_package(CXX11Features)
+
+#-----------------------------------------------------------------
+
+# create targets for (required and optional) dependencies to link to
+
+find_package(OpenMP)
+add_library(openmp INTERFACE)
+if (OPENMP_FOUND AND ERT_USE_OPENMP)
+    target_compile_options(openmp INTERFACE OpenMP_C_FLAGS)
+else ()
+     set(ERT_USE_OPENMP OFF)
      message(STATUS "OpenMP package not found - OpenMP disabled")
-  endif()
+endif ()
+
+add_library(pthread INTERFACE)
```

Comment:
It's a pretty cool cmake feature covered slightly in the blog I linked. From cmake doc:

> Creates an Interface Library. An INTERFACE library target does not directly create build output, though it may have properties set on it and it may be installed, exported and imported

What it does it that it creates a target (like any other `add_library` call) but *without any source files*. Then include dirs, libraries etc are set on this target, and it's used as any other `add_library`, which means we can link to it with `target_link_libraries`. What's neat about it is that it sets up *transitive* linking and include directories, so if `ecl` depends on `util`, and `util` depends on, say `lapack`, then `ecl` also depends on lapack *and* linking and include dirs are set up accordingly.